src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -1003,7 +1003,46 @@ private BetterResult BetterOperator(BinaryOperatorSignature op1, BinaryOperatorS
                 }
             }
 
-            return BetterResult.Neither;
+            // Always prefer operators with val parameters over operators with in parameters:
+            BetterResult valOverInPreference;
+
+            if (op1.LeftRefKind == RefKind.None && op2.LeftRefKind == RefKind.In)
```

Comment:
>if (op1.LeftRefKind == RefKind.None && op2.LeftRefKind == RefKind.In) [](start = 12, length = 69)

It feels like the logic here could be simplified if we add a helper that calculates the best among two ref kinds. Then we call it for both sides and then reason about final answer. 
Something like this:
```
BetterResult fromLeft = better(op1.LeftRefKind, op2.LeftRefKind);
BetterResult fromRight = better(op1.RightRefKind, op2.RightRefKind);
if (fromLeft == BetterResult.Neither || fromLeft == fromRight)
    return fromRight;
if (fromRight == BetterResult.Neither)
    return fromLeft;
return BetterResult.Neither;
```

Reply:
To me, the existing implementation is easier to read/trace. However, I don't feel strongly about this. I can implement your suggestions if you think it is the better alternative.

---
In reply to: [155026276](https://github.com/dotnet/roslyn/pull/23122#discussion_r155026276) [](ancestors = 155026276)